docs/js/doc-page.js

```diff
@@ -21,6 +21,94 @@ function removeDoubleSlashes(str) {
   return str.replace(/\/{2,}/g, '/');
 }
 
+function getPathFromSlug(str) {
+  return removeDoubleSlashes(`/${str}/`);
+}
+
+function relativeToAbsUrl(slug) {
+  return 'https://www.getambassador.io' + getPathFromSlug(slug);
+}
+
+const getDocPageSchema = ({
+  slug,
+  // an array of { title, slug } relative to parent pages (only if applicable)
+  // This doesn't include the root docs page nor the page itself
+  breadcrumbs = [],
+  // the doc page title (required)
+  title,
+  // the current docs version (required)
+  version,
+  // the current sidebar section (only if applicable)
+  section,
+  isFAQ,
+}) => ({
+  '@context': 'http://schema.org',
+  '@type': isFAQ ? 'FAQPage' : 'WebPage',
+  '@id': relativeToAbsUrl(slug),
+  // Not every search engine has assemblyVersion figured out (see below),
+  // we want to separate this doc page from others of different versions 😉
```

Comment:
Isn't it already separated by having a different `@id`?

Reply:
Indeed it's already separated, this is just an extra layer of cautiousness. If we didn't have this, search engines would understand the same page under different URLs (such as `/docs/latest/topics/concepts/` and `/docs/1.3/topics/concepts/`) as plain content duplication, which is treated quite severely.

With this property we're basically saying it's okay if most pages look mostly the same, they're part of the documentation for different versions of Ambassador. This can also potentially help search results for legacy users looking for "Service Preview Ambassador 1.3", for example. Whether or not Google and Bing will index the content this way is out of our control, though.

How do you suggest rephrasing this comment to make this distinction clearer, Luke? :)

Reply:
Perhaps:

Search engines normally penalize sites for duplicating content; we want to communicate to the search engine "we're hosting multiple versions of the docs, please don't penalize us for having content that is duplicated in multiple versions". Principally, we do that with `assemblyVersion` (below), but some search engines don't have `assemblyVersion` (the version of the software being described) figured out, so, we set `version` (the version of the web page describing the software) too.

Reply:
Sounds good! The discussion above on the `version` property will impact the final result of this comment, but I've copied it over locally in case we go through with it. Waiting on your answer above 😉